lib/iris/tests/integration/varying_references/test_roundtrip_time_varying_references.py

```diff
@@ -0,0 +1,276 @@
+        ]
+        reference_cube = Cube(
+            co_ref.points,
+            standard_name=co_ref.standard_name,
+            units=co_ref.units,
+            dim_coords_and_dims=[
+                (ref_dimcoord, i_refdim)
+                for i_refdim, ref_dimcoord in enumerate(ref_dimcoords)
+            ],
+            attributes=co_ref.attributes,
+        )
+        if not reference_cube.coords("time"):
+            # Add a dummy time coordinate to non-time-dependent reference cube
+            # - mostly because otherwise it cannot be saved to GRIB format
+            # NOTE: we give this a different nominal time to any of the data : when
+            # there is only one reference field, it's recorded time value should be
+            # **ignored** by the loader
+            reference_cube.add_aux_coord(
+                DimCoord(
+                    np.array(0, dtype=np.float32),
+                    standard_name="time",
+                    units="days since 1900-01-01",
+                )
+            )
+        cubes.append(reference_cube)
+
+    return cubes
+
+
+def check_expected(result_cubes, file_extension, time_dependence, zcoord_type):
+    assert len(result_cubes) == 2
+    result_phenom = result_cubes.extract_cube("air_temperature")
+
+    if zcoord_type == "pressure":
+        ref_coord_name = ref_cube_name = "surface_air_pressure"
+        if file_extension == "grib2":
+            ref_cube_name = "air_pressure"
+    elif zcoord_type == "height":
+        ref_coord_name = ref_cube_name = "surface_altitude"
+    else:
+        raise ValueError(f"Unknown hybrid coordinate type: {zcoord_type}")
+
+    result_ref_cube = result_cubes.extract_cube(ref_cube_name)
+    result_ref_coord = result_phenom.coord(ref_coord_name)
+
+    # Check that the reference cube and the coord are equivalent
+    assert result_ref_coord.shape == result_ref_cube.shape
+    assert np.array_equal(result_ref_cube.data, result_ref_coord.points)
+    assert not result_ref_coord.bounds  # bounds are unused in our testcases
+
+    # Check the expected phenomenon shape
+    if time_dependence == "static" and file_extension in ("pp", "grib2"):
+        phenom_shape = (NT, NZ, NY, NX)
+    else:
+        phenom_shape = (NZ, NT, NY, NX)
+    assert result_phenom.shape == phenom_shape
+
+    # Check expected reference values against calculated values.
+    # This shows that the reference was correctly divided into 2d fields and
+    # reconstructed on load to match the original (for fields-based formats).
+    if time_dependence == "static":
+        ref_shape = (NY, NX)
+    else:
+        ref_shape = (NT, NY, NX)
+    ref_data = 1000.0 + np.arange(np.prod(ref_shape)).reshape(ref_shape)
+    if zcoord_type == "pressure" and file_extension == "grib2":
+        # values come back in Pa not hPa
+        ref_data *= 100.0
+    assert np.array_equal(ref_data, result_ref_cube.data)
+
+
+_file_formats = ["pp", "nc"]
```

Comment:
I don't think so because usually in netcdf, a dataset is **_not_** split into separate fields, such as 2D slices, but retains all its necessary dimensions.
In principle, if we had different variables containing individual sections of data, with suitable additional scalar coords (e.g. time), then a merge would occur on load.  But this would be highly unusual, even if they were in separate files.
In practice, if we have data in different files it may be a candidate for _concatenate_, but I've never seen a case where a _merge_ can happen.

To be clear, I wouldn't expect any of the changes here to affect netcdf load/save in "normal" cases.
I only included "nc" as a format to confirm that it _doesn't_ impact the normal load/save operations.
The data generate by these tests will all save as netcdf in the "normal" way.

Reply:
Ok I added  https://github.com/SciTools/iris/pull/6168/commits/cb1ddff49e042705c221476137b81ee88033de9a to cover that idea.

It splits the test-data into 2D fields + saves each to a different file.
The result combines ok as long as concatenation is enabled from the outset (the auto-detect doesn't work because the multiple reference cubes can concat but not merge).

I am increasingly impressed by concatenate : it can do a lot more than I realised.